### ISSUE TITLE
Why is $this->clients so large?

### ISSUE THREAD

**Issue by fx1234** [ratchetphp/Ratchet#802]
Hi

i just startet with ratchet and i did some basic testes. i used the example from the tutorial page (http://socketo.me/docs/hello-world).

to understand what _**$this->clients**_ contains, i added a _**var_dump($this->clients)**_ to the onOpen()-method. 

than i started the server with '_**php chat-server.php >output.txt**_'

with one client, the output.txt is _600 KB_, with two clients its _20 MB_, with three its _200 MB_ and with four clients its _800 MB_ (yes, MB).

is this normal? why is  $this->clients so big?

**Comment by cboden** [comment]
I believe there's a cyclic reference. Every new could would add a couple new bytes. Check the memory footprint. By var_dumping the contents of an object with a cyclic reference the logger is recursing which is ballooning your logs. PHP can handled cyclic references just fine, logging, not so much.

**Comment by WyriHaximus** [comment]
TL;DR It's not large, the references it makes look massive

As @cboden mentions, every connection has a reference to the event loop, and that holds references to all open connections, timer listeners, ticks, signals listeners, and everything all of those objects and callables hold references to. So @fx1234 you can var_dump it but then you get everything that references everything that references everything until you hit a limit and PHP cuts you out.

**Comment by fx1234** [comment]
thanks guys

I was hoping it would be something like this, but I wasn't sure

is there any documentation/information about $this->clients (ConnectionInterface $conn) and what data it contains?

**Comment by cboden** [comment]
`$clients` is an object you created, as per the tutorial. It is not part of Ratchet. In the tutorial it's just a container that holds all the active connections.

**Comment by fx1234** [comment]
And where are all the data in _**ConnectionInterface $conn**_ comming from?

**Comment by cboden** [comment]
http://socketo.me/docs/design
